### ISSUE TITLE
Context Menus break Auto Layout

### ISSUE THREAD

**Issue by alexeyprimechaev** [apptekstudios/ASCollectionView#77]
**Preface**

Not really this library's problem, but it _could_ help solve it by bringing in support for UIKit Context Menu as opposed to relying on SwiftUI's problematic implementation.

**Issue**

The issue is as follows: whenever a Context Menu is accessed, Xcode throws an error regarding some broken constraints — on any view in any configuration (screenshots attached), but sometimes a context menu preview is recovered by being placed on top of the source view.

If, however, a view has dynamic content/Button in it, it's simply not displayed.

**Proposed Solution**

Implement per-cell Context Menus using UICollectionView API.

I think that you have created a fantastic framework that is simply necessary to create any sort of a complex and dynamic user interface in SwiftUI (at least until Apple adds a CollectionView analog) and it would greatly increase its value by providing a solution to this issue. _**Pretty Please?**_

**To Reproduce**

Interact with a Context Menu

**Screenshots**

Code
<img width="260" alt="Screen Shot 2020-01-14 at 11 58 31 PM" src="https://user-images.githubusercontent.com/51163708/72382181-c8300b00-3729-11ea-8d4e-a02bf2f981b7.png">

Rendered Result
![One](https://user-images.githubusercontent.com/51163708/72382028-78e9da80-3729-11ea-9eeb-f391ac4f111c.jpeg)

Example with dynamic content/Button
![IMG_1E51CA355FDF-1](https://user-images.githubusercontent.com/51163708/72382041-80a97f00-3729-11ea-8677-d9c74691dfac.jpeg)

Xcode Output
<img width="790" alt="image" src="https://user-images.githubusercontent.com/51163708/72382464-5e643100-372a-11ea-9a24-40d5aad91894.png">

**Xcode Version:**

13.3.1

**Simulator, Device, Both?**

Both

**Comment by 1234igor** [comment]
+1

Struggling with same issue and don’t know what should I do.

**Comment by apptekstudios** [comment]
I'm currently overseas but will have a look into this as soon as I can! Should be relatively straightforward to implement support via UIKit :)

**Comment by apptekstudios** [comment]
@alexeyprimechaev @1234igor 

I've made a start by adding support for `UIContextMenuConfiguration` [to the dev branch](https://github.com/apptekstudios/ASCollectionView/tree/dev). See the 'magazineLayout' demo page for an example 🎉.

We could wrap it further to make a clearer struct-based API, however the UIKit method is relatively straightforward and allows for complexities such as nested menus - so I'll leave it be for now 👍 

Interestingly, during my testing I found that constraints were broken even in a pure UIKit Collectionview using contextMenus (with no ASCollectionView or swiftUI involved). So sounds like there's also a UIKit bug :/ gladly it's just an annoying log message!
